src/java/grl/src/grl/driver/GRL_Driver_ChangeModes.java

```diff
@@ -0,0 +1,1144 @@
+    
+    private static final int MILLI_SLEEP_TO_EMULATE_COMPUTATIONAL_EFFORT = 40;
+    
+    final boolean debugPrintoutFlag = false;
+    
+    // Tool Data
+    private static final String TOOL_FRAME = "toolFrame";
+    private static final double[] TRANSLATION_OF_TOOL = { 0, 0, 100 };
+    private static final double MASS = 0;
+    private static final double[] CENTER_OF_MASS_IN_MILLIMETER = { 0, 0, 100 };
+    
+   // private FlexFellowIOGroup _flexFellowIOGroup;
+	
+	@Override
+	public void initialize()
+	{
+		_startStopUI = new StartStopSwitchUI(this);
+		_processDataManager = new ProcessDataManager(this);
+		_lbrController = (Controller) getContext().getControllers().toArray()[0];
+		_lbr = (LBR) _lbrController.getDevices().toArray()[0];
+		
+		//if(_flexFellowPresent) _flexFellowIOGroup = new FlexFellowIOGroup(_lbrController);
+//		if(_mediaFlangeIOPresent) {
+//			_mediaFlangeIOGroup = new MediaFlangeIOGroup(_lbrController);
+//		//	_mediaFlangeIOGroup.setLEDBlue(true);
+//		}
+
+        _friConfiguration = FRIConfiguration.createRemoteConfiguration(_lbr, _processDataManager.get_FRI_KONI_LaptopIPAddress());
+        _friConfiguration.setSendPeriodMilliSec(4);
+
+        _friSession = new FRISession(_friConfiguration);
+
+		_flangeAttachment = getApplicationData().createFromTemplate("UltrasoundCamera");
+		_pausedApplicationRecovery = getRecovery();
+
+		LoadData _loadData = new LoadData();
+		_loadData.setMass(_processDataManager.getEndEffectorWeight());
+		_loadData.setCenterOfMass(_processDataManager.getEndEffectorX(),
+				_processDataManager.getEndEffectorY(),
+				_processDataManager.getEndEffectorZ());
+		//_toolAttachedToLBR = new Tool("Tool", _loadData);
+		// TODO: make configurable via C++ API?
+		_toolAttachedToLBR = _flangeAttachment;
+		_toolAttachedToLBR.attachTo(_lbr.getFlange());
+
+		_jointLimits = _lbr.getJointLimits();
+		
+		// used when setting limits in _HandGuidingMotion
+		_maxAllowedJointLimits = _jointLimits.getMaxJointPosition().get();
+		_minAllowedJointLimits = _jointLimits.getMinJointPosition().get();
+		for (int i = 0; i < _lbr.getJointCount(); i++) {
+			_maxAllowedJointLimits[i] -= 0.05;
+			_minAllowedJointLimits[i] += 0.05;
+		}
+
+
+		_teachControlMode = new JointImpedanceControlMode(_lbr.getJointCount())
+								.setStiffnessForAllJoints(0.1)
+								.setDampingForAllJoints(0.7);
+	
+
+		_teachModeRunnable = new TeachMode(
+				_flangeAttachment,
+				_maxAllowedJointLimits,
+				_minAllowedJointLimits);
+		_teachModeRunnable.setLogger(getLogger());
+		_teachModeThread = new Thread(_teachModeRunnable);
+		_teachModeThread.start();
+	}
+	
+
+	@Override
+	public void run()
+	{
+
+		getLogger().info("GRL_Driver from github.com/ahundt/grl starting...\nZMQ Connecting to: " + _processDataManager.get_ZMQ_MASTER_URI());
+
+		// connect to the controlling application via ZeroMQ
+		ZMQManager zmq = new ZMQManager(_processDataManager.get_ZMQ_MASTER_URI(),getLogger());

```

Comment:
Will this conflict with @Ashkan372's pull request #107? If not, will need to make the change to reconcile the differences! Perhaps it is designed to abstract the difference away across an interface?